watertap/unit_models/reverse_osmosis_base.py

```diff
@@ -0,0 +1,789 @@
+                    [p for p in self.config.property_package.phase_list]
+                )
+            )
+
+        validate_membrane_config_args(self)
+
+        self._add_feed_side_membrane_channel_and_geometery()
+
+        self.feed_side.add_state_blocks(has_phase_equilibrium=False)
+
+        self.feed_side.add_material_balances(
+            balance_type=self.config.material_balance_type, has_mass_transfer=True
+        )
+
+        self.feed_side.add_energy_balances(
+            balance_type=self.config.energy_balance_type, has_enthalpy_transfer=True
+        )
+
+        self.feed_side.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            pressure_change_type=self.config.pressure_change_type,
+            has_pressure_change=self.config.has_pressure_change,
+        )
+
+        self.feed_side.add_isothermal_conditions()
+
+        self.feed_side.add_volumetric_flowrate_balance()
+
+        self.feed_side.add_concentration_polarization(
+            concentration_polarization_type=self.config.concentration_polarization_type,
+            mass_transfer_coefficient=self.config.mass_transfer_coefficient,
+        )
+
+        self.feed_side.apply_transformation()
+
+        self.feed_side.add_expressions()
+
+        add_object_reference(self, "length_domain", self.feed_side.length_domain)
+        add_object_reference(
+            self, "difference_elements", self.feed_side.difference_elements
+        )
+        add_object_reference(self, "first_element", self.feed_side.first_element)
+        add_object_reference(self, "nfe", self.feed_side.nfe)
+
+        _add_object_reference_if_exists(self, "dP_dx", self.feed_side, "dP_dx")
+        _add_object_reference_if_exists(self, "area_cross", self.feed_side, "area")
+
+        _add_object_reference_if_exists(
+            self, "cp_modulus", self.feed_side, "cp_modulus"
+        )
+        _add_object_reference_if_exists(self, "Kf", self.feed_side, "K")
+        _add_object_reference_if_exists(self, "Kf_avg", self.feed_side, "K_avg")
+        _add_object_reference_if_exists(
+            self, "channel_height", self.feed_side, "channel_height"
+        )
+        _add_object_reference_if_exists(self, "dh", self.feed_side, "dh")
+        _add_object_reference_if_exists(
+            self, "spacer_porosity", self.feed_side, "spacer_porosity"
+        )
+        _add_object_reference_if_exists(self, "N_Re", self.feed_side, "N_Re")
+        _add_object_reference_if_exists(self, "N_Re_avg", self.feed_side, "N_Re_avg")
+        _add_object_reference_if_exists(self, "N_Sc", self.feed_side, "N_Sc")
+        _add_object_reference_if_exists(self, "N_Sh", self.feed_side, "N_Sh")
+        _add_object_reference_if_exists(self, "velocity", self.feed_side, "velocity")
+        _add_object_reference_if_exists(
+            self, "friction_factor_darcy", self.feed_side, "friction_factor_darcy"
+        )
```

Comment:
This is all for backwards compatibility right? I think that's a good approach for this PR, but in the near future we may want to change our tests to the new format and not support this. At least for WaterTAP we don't have many external users who would be mad about this breaking change.

Reply:
Yes. I agree we should probably break it, and should be in a separate PR as it requires touching a huge number of files.